### PR TITLE
Region status Adding and Removing

### PR DESCRIPTION
**Pull request by liyuheng55555**

### CODE REVIEW

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/region/AddRegionPeerProcedure.java [hunk 1]
```diff
@@ -92,6 +94,7 @@ protected Flow executeFromState(ConfigNodeProcedureEnv env, AddRegionPeerState s
           setNextState(AddRegionPeerState.DO_ADD_REGION_PEER);
           break;
         case DO_ADD_REGION_PEER:
+          handler.updateRegionCache(consensusGroupId, destDataNode, RegionStatus.Adding);
```
Comment (CRZbulabula) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/region/AddRegionPeerProcedure.java:97]: Why should update RegionStatus to Adding twice?

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/region/RemoveRegionPeerProcedure.java [hunk 2]
```diff
@@ -100,6 +102,7 @@ protected Flow executeFromState(ConfigNodeProcedureEnv env, RemoveRegionPeerStat
           setNextState(DELETE_OLD_REGION_PEER);
           break;
         case DELETE_OLD_REGION_PEER:
+          handler.updateRegionCache(consensusGroupId, targetDataNode, RegionStatus.Removing);
```
Comment (CRZbulabula) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/region/RemoveRegionPeerProcedure.java:105]: Why should update RegionStatus to Removing twice?

**Review by CRZbulabula — COMMENTED** [review]
PTAL~

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/region/AddRegionPeerProcedure.java (hunk 1 above)
Comment (liyuheng55555) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/region/AddRegionPeerProcedure.java:97]: This is for robustness. The procedure may recover to different state.

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/region/RemoveRegionPeerProcedure.java (hunk 2 above)
Comment (liyuheng55555) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/region/RemoveRegionPeerProcedure.java:105]: same as above

**Review by CRZbulabula — APPROVED** [review]
LGTM

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**